tests/rptest/tests/datalake/iceberg_toggling_test.py

```diff
@@ -0,0 +1,173 @@
+                last_count = dl.spark().count_table("redpanda",
+                                                    self.topic_name)
+
+                while time.time() - start_time < timeout_sec:
+                    current_count = dl.spark().count_table(
+                        "redpanda", self.topic_name)
+                    self.logger.debug("Current count: {current_count}")
+
+                    if current_count == last_count:
+                        # If no more records were added and the proof interval
+                        # has passed then we can consider the iceberg quiesced.
+                        if last_time + proof_interval_sec < time.time(
+                        ) and current_count >= min_records:
+                            self.logger.info(
+                                f"Iceberg quiesced in {time.time() - start_time} with {current_count} records"
+                            )
+                            return current_count
+                    else:
+                        # Reset the timer and count.
+                        self.logger.debug(
+                            f"Records added: {current_count - last_count}")
+                        last_time = time.time()
+                        last_count = current_count
+                    # Don't spin too fast
+                    time.sleep(1)
+
+                assert last_count >= min_records, f"Timed out waiting for minimum records in iceberg {last_count=} < {min_records=}"
+                assert False, "Iceberg did not quiesce in time"
+
+            for ix in range(1, NUM_TOGGLING_ITERATIONS + 1):
+                self.logger.info(f"Starting toggling iteration {ix}")
+
+                # Wait for some records to be produced.
+                ensure_stream_progress(total_records_sent() +
+                                       ADDITIONAL_RECORDS_PER_ITERATION)
+
+                # Wait for records to be translated.
+                min_records_expected = total_records_sent()
+                dl.wait_for_translation(self.topic_name,
+                                        min_records_expected,
+                                        op=operator.gt)
+
+                self.rpk.alter_topic_config(self.topic_name,
+                                            TopicSpec.PROPERTY_ICEBERG_MODE,
+                                            "disabled")
+
+                wait_iceberg_quiesced(min_records=min_records_expected)
+
+                self.rpk.alter_topic_config(self.topic_name,
+                                            TopicSpec.PROPERTY_ICEBERG_MODE,
+                                            "key_value")
+
+            # We're done, let's stop the stream.
+            connect.stop_stream(name=self.stream_name, should_finish=None)
+
+            # Wait for everything to be translated.
+            total_row_count = total_records_sent()
+            dl.wait_for_translation(self.topic_name, total_row_count)
```

Comment:
Curious did we need the op to be >= here? (I don't think so, but just asking given the first commit)

Reply:
Not here. Here we stopped the producer so we know the exact count to expect. I'm using `operator.ge` above though.